plugins/inputs/mqtt_consumer/mqtt_consumer.go

```diff
@@ -240,11 +250,25 @@ func compareTopics(expected []string, incoming []string) bool {
 }
 
 func (m *MQTTConsumer) onMessage(acc telegraf.TrackingAccumulator, msg mqtt.Message) error {
+	m.bytesRecv.Set(0)
 	metrics, err := m.parser.Parse(msg.Payload())
 	if err != nil {
 		return err
 	}
 
+	if m.collectBytes {
+		byteCount := unsafe.Sizeof(msg) +
+			unsafe.Sizeof(msg.Payload()) +
+			unsafe.Sizeof(msg.MessageID()) +
+			unsafe.Sizeof(msg.Duplicate()) +
+			unsafe.Sizeof(msg.Qos()) +
+			unsafe.Sizeof(msg.Topic()) +
+			unsafe.Sizeof(msg.Retained())
+		byteCount64 := int64(byteCount)
+		m.bytesRecv.Incr(byteCount64)
```

Comment:
Can't this be combined to not have an extra allocation?
```suggestion
		m.bytesRecv.Incr(int64(byteCount))
```